Changes to make main branch become 8.1

joperezr opened the pull request:
This PR is switching main to become the next 8.1 branch. Before this change, main branch used to be our 9.0 branch, but given this repo intends to ship monthly releases and wants those monthly releases the main target of development work, we decided to switch main branch to become 8.1, and to create a new `dev` branch which would become our 9.0 branch. In order to do all this, I followed this process:

1. First we merged all of the changes that were made in release/8.0 branch back to main.
2. Then we created a new `dev` branch from the tip of main, which will become our new 9.0 branch. Work that is 9.0-specific will go into this branch going forward. Here is a link to that branch: https://github.com/dotnet/extensions/tree/dev
3. Make it so `main` branch is the new 8.1 branch by having its contents completely match what we have in release/8.0 branch as we prepared for the release. This is the work included in this PR. To do this, I basically checked out main locally, deleted all of its contents and then did copy paste of all of the contents of a fresh clone of extensions release/8.0 branch. Doing so generated the changes (diff between main and release/8.0) included in this PR. Given we started of with step 1 above, it means that the diff generated on this PR is strictly changes that were done in `main` which were not in release/8.0 (either intentionally by making changes into main, or issues when resolving merge conflicts). The intention is to merge this PR as is, as this is the state we want for release/8.1. We can also use the diffs generated by this PR to fix bad merges that may have happened in `dev` branch.
4. Added one more commit here to do branding updates for 8.1.
5. Once this is merged, we will want to update dependency flow and subscriptions so that they match the new branching scheme we plan to use.

cc: @RussKie @Tratcher @sebastienros @geeknoid @martintmk @xakep139 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4695)

RussKie commented on docs/building.md:
```diff
@@ -45,8 +45,8 @@ Here are few commands that you will likely use the most:
     - `build.sh --restore`: to install the required .NET SDK, .NET tools and the toolset. This is equivalent to running `./restore.sh`.
     - `build.sh --build`: to build the solution<sup>1</sup>.
     - `build.sh --test`: to run all unit tests in the solution<sup>1</sup>.
-    - `build.sh --vs <keywords>`: to generate a "filtered" solution and save it as `SDK.sln`. It also performs the "restore" operation. Keywords can be any part of the name or path of project files you want to include. For example: `./build.sh --vs Http,Fakes,AspNetCore`.<br />
```
This line should be reverted

RussKie commented on eng/common/build.sh:
```diff
@@ -105,7 +105,7 @@ while [[ $# > 0 ]]; do
       binary_log=true
       ;;
-    -excludecibinarylog|-nobl)
+    -excludeCIBinarylog|-nobl)
```
I don't think this is correct either, all `$opt` should be in the lower case

joperezr replied:
Given this is part of eng\common and controlled by arcade, I'll leave as is.

RussKie commented on global.json:
```diff
@@ -1,24 +1,22 @@
-    "Microsoft.DotNet.Arcade.Sdk": "9.0.0-beta.23557.7",
-    "Microsoft.DotNet.Helix.Sdk": "9.0.0-beta.23557.7"
+    "Microsoft.DotNet.Arcade.Sdk": "8.0.0-beta.23525.4",
+    "Microsoft.DotNet.Helix.Sdk": "8.0.0-beta.23525.4"
```
I'd keep the newer Arcade, if possible

joperezr replied:
9.0 Arcade will start depending on the 9.0 SDK soon, and this branch should only consume 8.0 things, so I've made this branch subscripte t o 8.0 band SDK updates like other dotnet servicing branches.

RussKie commented on .vscode/settings.json:
```diff
@@ -1,3 +0,0 @@
-{
```
This shouldn't be deleted

RussKie commented on src/Analyzers/Microsoft.Analyzers.Local/Microsoft.Analyzers.Local.csproj:
```diff
@@ -30,10 +28,6 @@
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" />
     <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" PrivateAssets="all" IncludeAssets="runtime; build; native; contentfiles; analyzers; buildtransitive" />
-
-    <PackageReference Include="System.Text.Json" PrivateAssets="all" GeneratePathProperty="true" />
-    <None Include="$(OutputPath)\$(AssemblyName).dll" Pack="true" PackagePath="analyzers/dotnet/cs" Visible="false" />
-    <None Include="$(PkgSystem_Text_Json)\lib\netstandard2.0\*.dll" Pack="true" PackagePath="analyzers/dotnet/cs" Visible="false" />
```
Doesn't this indicate the changes in main are an improvement over the 8.0 branch? Should those be retained?

joperezr replied:
I'm honestly not sure about this one. @geeknoid any clues on this one?

RussKie commented on src/Libraries/Microsoft.Extensions.Diagnostics.ExceptionSummarization/ExceptionSummarizer.cs:
```diff
@@ -31,9 +31,8 @@ public ExceptionSummarizer(IEnumerable<IExceptionSummaryProvider> providers)
-
         var exceptionType = exception.GetType();
-        var exceptionTypeName = exceptionType.Name;
+        var exceptionTypeName = exception.GetType().Name;
```
ditto

RussKie commented on src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/Enrichment/ILogEnricher.cs:
```diff
@@ -4,7 +4,7 @@
 
 /// <summary>
-/// Represents a component that augments log records with additional properties.
+/// Augments log records with additional properties.
```
The xml-docs should be preserved as those were an improvement in the main, see https://github.com/dotnet/extensions/pull/4393.

RussKie commented on src/Libraries/Microsoft.Extensions.Telemetry/Microsoft.Extensions.Telemetry.json:
```diff
@@ -296,4 +296,4 @@
     }
   ]
-}
+}
```
This can be reverted.

RussKie commented on src/Shared/Pools/PooledListPolicy.cs:
```diff
@@ -4,6 +4,7 @@
 using System.Collections.Generic;
```
revert

RussKie commented on test/Analyzers/Microsoft.Analyzers.Extra.Tests/CallAnalysis/ArraysTests.cs:
```diff
@@ -122,7 +122,7 @@ public void Triggers()
                 Assembly.GetAssembly(typeof(FrozenDictionary<,>))!,
             },
-            new[] { Source });
+            new[] { Source }).ConfigureAwait(false);
```
revert

RussKie commented on test/Analyzers/Microsoft.Analyzers.Extra.Tests/CallAnalysis/LegacyCollectionTests.cs:
```diff
@@ -44,7 +44,7 @@ public void Foo()
                 Assembly.GetAssembly(typeof(System.Collections.Specialized.HybridDictionary))!
             },
-            new[] { Source });
+            new[] { Source }).ConfigureAwait(false);
```
revert

RussKie commented on test/Analyzers/Microsoft.Analyzers.Extra.Tests/CallAnalysis/LegacyLoggingTests.Extra.cs:
```diff
@@ -92,7 +92,7 @@ static partial class Log
             new LegacyLoggingFixer(),
             new[] { Assembly.GetAssembly(typeof(ILogger))!, Assembly.GetAssembly(typeof(LoggerMessageAttribute))! },
-            new[] { OriginalSource, OriginalTarget });
+            new[] { OriginalSource, OriginalTarget }).ConfigureAwait(false);
```
revert all of these

RussKie commented on test/Analyzers/Microsoft.Analyzers.Extra.Tests/Resources/RoslynTestUtils.cs:
```diff
@@ -146,7 +146,7 @@ public static void AssertDiagnostic(this string text, int spanNum, DiagnosticDes
         catch (ArgumentOutOfRangeException)
         {
-            Assert.Fail($"Unexpected warning {actual}");
+            Assert.True(false, $"Unexpected warning {actual}");
```
revert

RussKie commented on test/Libraries/Microsoft.AspNetCore.HeaderParsing.Tests/ParserTests.cs:
```diff
@@ -76,9 +76,7 @@ public void Cookkie_ReturnsParsedValue()
         Assert.True(CookieHeaderValueListParser.Instance.TryParse(sv, out var result, out var error));
-#pragma warning disable xUnit2013 // Do not use equality check to check for collection size.
         Assert.Equal(1, result.Count);
-#pragma warning restore xUnit2013 // Do not use equality check to check for collection size.
```
revert all these

joperezr replied:
If we are not using the new tooling though, I'm guessing these are not needed? Especially since they are just trying to silence an analyzer.

RussKie commented on test/Libraries/Microsoft.Extensions.Diagnostics.Testing.Tests/Metrics/MetricCollectorTests.cs:
```diff
@@ -413,6 +413,6 @@ public static void Snapshot()
         snap = collector.GetMeasurementSnapshot(true);
         Assert.Equal(3, snap.Count);
-        Assert.Empty(collector.GetMeasurementSnapshot());
+        Assert.Equal(0, collector.GetMeasurementSnapshot().Count);
```
revert

RussKie commented on test/Libraries/Microsoft.Extensions.Telemetry.Abstractions.Tests/Logging/LoggerMessageStateTests.cs:
```diff
@@ -129,7 +129,6 @@ public static void CollectorContract()
     [Fact]
     public static void ReadOnlyListContract()
     {
-#pragma warning disable xUnit2013 // Do not use equality check to check for collection size.
```
revert

joperezr commented:
Thanks for all of the great feedback @RussKie 😃. The intent of this PR is to have it exactly match the current contents of release/8.0 (except for branding changes as we don't want builds to clash) so I'll go ahead and merge this PR as is, and will submit a follow up PR addressing all of your comments where those could be better looked at and reviewed individually. I hope that's okay with you 😃. The main rush for merging this is so that a) we don't block progress for 8.1, and b) we don't get more conflicts due to new changes coming into main branch.

joperezr commented:
Created #4705 which addresses the feedback from this PR.

RussKie commented:
@joperezr this doesn't look right to me:
```
PS> darc get-default-channels --source-repo "https://github.com/dotnet/extensions"  --verbose

(931)  https://github.com/dotnet/extensions @ release/3.1 -> .NET Core 3.1 Release
(5039) https://github.com/dotnet/extensions @ release/8.0 -> .NET 8
(5038) https://github.com/dotnet/extensions @ main -> .NET 9
```

The `dev` should be publishing to `.NET 9`. Should the `main` be publishing to `.NET 8`?
